### PR TITLE
feat(TDP-4661): add quality filter to column header

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4661

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/5/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/5/)

**Comment by frassinier** [comment]
image

Looks weird, badge are not translated :|

File: dataprep-webapp/src/app/components/datagrid/header/datagrid-header.html [hunk 1]
```diff
@@ -61,7 +66,7 @@
     </empty-menu-items>
 
     <invalid-menu-items>
-        <li ng-click="datagridHeaderCtrl.addFilter('invalid_records')">
+        <li ng-click="datagridHeaderCtrl.addFilter(datagridHeaderCtrl.INVALID_RECORDS)">
```
[the remark this answers is not in the document]
Comment (charlesnguyen) [reply dataprep-webapp/src/app/components/datagrid/header/datagrid-header.html]: they are used in the line 27.

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/8/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/8/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/14/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/14/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/15/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/15/)

**Pull request merged**